### ISSUE TITLE
Trakt addon not syncing watched tv shows and movies anymore

### ISSUE THREAD

**Issue by monarc83** [razzeee/script.trakt#429]
Hi all,

Since a couple of days, I am not sure what I have changed, maybe it was the upgrade to the latest official Kodi release, the Trakt addon isn't working anymore.

`23:24:59.339 T:138228761680 WARNING: [script.trakt] trakt.interfaces.base: Request failed: "GET /shows/74543/seasons/6/episodes/3" - 404: "Method exists, but no record found" (Not Found)
23:24:59.340 T:138228761680   FATAL: [script.trakt] resources.lib.service: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'AttributeError'>
                                            Error Contents: ("'NoneType' object has no attribute 'to_dict'",)
                                            Traceback (most recent call last):
                                              File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/script.trakt/resources/lib/service.py", line 42, in _dispatch
                                                self.scrobbler.playbackStarted(data)
                                              File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/script.trakt/resources/lib/scrobbler.py", line 181, in playbackStarted
                                                self.curVideoInfo['number']).to_dict()}
                                            AttributeError: 'NoneType' object has no attribute 'to_dict'
                                            -->End of Python script error report<--`

It would be great if somebody has an idea how to fix the issue? 

Many thanks in advance.

Best Regards,
Marc

**Comment by razzeee** [comment]
https://github.com/trakt/script.trakt#i-found-something-that-doesnt-work

**Comment by monarc83** [comment]
Hi,

Thanks for the reply. I was hoping the log snip above may point out the issue already. I will upload the more detailed log later today. 

in case there is already a same issue tracked here, it would be nice if you can share the url or #, since I haven't found it while I was searching.

BR,
Marc

**Comment by monarc83** [comment]
Attached now the kodi log with Kodi and the trakt addon in debug mode.


[kodi.log](https://github.com/trakt/script.trakt/files/1720105/kodi.log)

**Comment by monarc83** [comment]
Have somebody had the chance to look into the debug logs? Anything else I can share?

It would be great to know the root cause of this issue and how to fix it.

BR,
Marc

**Comment by razzeee** [comment]
Please see https://github.com/trakt/script.trakt#creating-logfiles on how to create a logfile.

There were also some weird error messages from your scraper, can you make sure you have the newest version of the tvdb scraper?

**Comment by chriskol** [comment]
I'll try to get a log file up later today but I also get the `EXCEPTION Thrown (PythonToCppException)` error.  No authorization dialog comes up.  Username is blank.  Clicking the 'how do i auth...' text generates this error as well.

**Comment by chriskol** [comment]
Here's my [log file](http://paste.kodi.tv/udopagivif). Got a little messy at the end as I realized I didn't have proper log uploader installed so I had to do that.  The log also encompasses me installing Trakt (after having uninstalled all orphaned dependancies) to make it as straightforward as possible.

If it helps, I'd authorized Trakt through Incursion addon earlier and the process worked fine.

**Comment by razzeee** [comment]
It does not like your dateutil installation. And you did not enable debug logging, see the readme at the top of the repo pls.

**Comment by Razzeee** [comment]
It does not like your dateutil installation. And you did not enable debug logging, see the readme at the top of the repo pls.

**Comment by chriskol** [comment]
Debug logging was enabled unless I enabled the wrong log options in Kodi (though the status info in the top left of the screen was running) and the wrong one in trakt (though there's only one option in trakt).  Other thing maybe is I installed the trakt addon while logging was happening so the logging was enabled because of remembering previous settings?  I'll try turning off logging in trakt, uninstalling the addon and orphaned dependencies, then creating a new log from startup, installing trakt, then turning logging on and trying to authorize (assuming it doesn't work before then).

I'm not much of a developer, much less a python guy, but what can I do about dateutil?  The logs seem to reference higher up that it is installed, but then the addon says it isn't later?  If I can uninstall it, will it be reinstalled as a dependency when I install the trakt addon?

**Comment by razzeee** [comment]
yes, if you manage to uninstall it, kodi will reinstall it on next install.

This is the info I was talking about earlier https://github.com/trakt/script.trakt#creating-logfiles

**Comment by chriskol** [comment]
I couldn't delete dateutil without getting into some dependency mess where I'd have to uninstall a bunch of other stuff so I forced an update (even though it was the same version).  I re-installed the Trakt addon and it works now!  Thanks for everything!  Not sure if this is the same experience as the original poster but my issue is resolved!

**Comment by monarc83** [comment]
Sounds great Chris. I have immediately checked my Kodi installs but couldn't find dateutil under addons. Could you please let me know where to find it. Many thanks in advance.

**Comment by chriskol** [comment]
@monarc83 I'm away from my setup at the moment, but the page you're looking for is 'manage dependencies'.  I think it's under 'system'.  You might need to be in Expert mode.  I uninstalled the Trakt addon, then went into 'manage dependencies', uinstalled trakt.py, then went to 'dateutil' and ran an update from the official repo.  When I reinstalled the Traket addon, the auth screen came up immediately.
